rpc/errors.go

```diff
@@ -199,3 +257,51 @@ var (
 		Message: "An unexpected error occurred",
 	}
 )
+
+type ContractErrorData struct {
+	// the execution trace up to the point of failure
+	RevertError ContractExecutionError `json:"revert_error,omitempty"`
+}
+
+type TransactionExecutionErrorData struct {
+	// The index of the first transaction failing in a sequence of given transactions
+	TransactionIndex int `json:"transaction_index,omitempty"`
+	// the execution trace up to the point of failure
+	ExecutionError ContractExecutionError `json:"execution_error,omitempty"`
+}
+
+type ContractExecutionError struct {
+	// the error raised during execution
+	Message                      string `json:",omitempty"`
+	ContractExecutionErrorStruct `json:",omitempty"`
+}
+
+func (contractEx *ContractExecutionError) UnmarshalJSON(data []byte) error {
+	var contractErrStruct ContractExecutionErrorStruct
+	var message string
+
+	if err := json.Unmarshal(data, &message); err == nil {
+		*contractEx = ContractExecutionError{
+			Message:                      message,
+			ContractExecutionErrorStruct: contractErrStruct,
+		}
+		return nil
+	}
+
+	if err := json.Unmarshal(data, &contractErrStruct); err == nil {
+		*contractEx = ContractExecutionError{
+			Message:                      "",
+			ContractExecutionErrorStruct: contractErrStruct,
+		}
+		return nil
+	}
+
+	return fmt.Errorf("failed to unmarshal ContractExecutionError")
+}
+
+type ContractExecutionErrorStruct struct {
+	ContractAddress *felt.Felt              `json:"contract_address"`
+	ClassHash       *felt.Felt              `json:"class_hash"`
+	Selector        *felt.Felt              `json:"selector"`
+	Error           *ContractExecutionError `json:"error"`
+}
```

Comment:
Are we missing the "error message" field (seems like the spec is missing the property field?), and these should be omitempty since there's a oneOf condition?